cpmpy/solvers/pysat.py

```diff
@@ -255,18 +266,25 @@ def transform(self, cpm_expr):
 
             See the :ref:`Adding a new solver` docs on readthedocs for more information.
 
+
+            In the case of PySAT, the supported constraints are over Boolean variables:
+
+            - Boolean clauses
+            - Cardinality constraint (`sum`)
+            - Pseudo-Boolean constraints (`wsum`)
+
             :param cpm_expr: CPMpy expression, or list thereof
             :type cpm_expr: Expression or list of Expression
 
             :return: list of Expression
         """
         cpm_cons = toplevel_list(cpm_expr)
-        cpm_cons = decompose_in_tree(cpm_cons)
+        cpm_cons = decompose_in_tree(cpm_cons, supported=frozenset({"alldifferent"}))
         cpm_cons = simplify_boolean(cpm_cons)
-        cpm_cons = flatten_constraint(cpm_cons)
+        cpm_cons = flatten_constraint(cpm_cons)  # flat normal form
         cpm_cons = only_bv_reifies(cpm_cons)
         cpm_cons = only_implies(cpm_cons)
-        cpm_cons = canonical_comparison(cpm_cons)
+        cpm_cons = linearize_constraint(cpm_cons, supported=frozenset({"sum","wsum", "and", "or", "bv"}))  # the core of the MIP-linearization
```

Comment:
right, so I understand you added it for and/or, but I dont think "bv" should be explicitly listed?

Reply:
Ah, indeed should be removed (no longer uesd in linearize since boolvars are now properly linearized for each solver)